backend/LeetWars.Builder/Helpers/DirectoryConfigurations/DirectoryConfigurations.cs

```diff
@@ -0,0 +1,54 @@
+﻿using LeetWars.Builder.Models;
+
+namespace LeetWars.Builder.Helpers.DirectoryConfigurations;
+
+public class DirectoryConfigurations
+{
+    public string? CurrentDirectory { get; set; }
+
+    public async Task<string> CreateDirectoryCSharp(string directoryPath, CodeRunRequest request)
```

Comment:
too long method, separate logic and configuration into smaller methods

Reply:
elaborate please

Reply:
some methods you could implement: 

- private async Task WriteTextToFileAsync(string filePath, string content)
- private string GetRunnerDirectory(string directoryPath, string language, CodeRunRequest request)
- private string GetCSharpProjectFileContent()
- private string GetProjectFileName(CodeRunRequest request)

and so on 

this way CreateDirectoryCSharp method will be significantly more readable